src/compiler/compiler.rs

```diff
@@ -1082,32 +1083,39 @@ where
     // Both clang and clang-cl define _MSC_VER on Windows, so we first
     // check for MSVC, then check whether _MT is defined, which is the
     // difference between clang and clang-cl.
-    let test = b"#if defined(__NVCC__)
-nvcc
+    //
+    // We prefix the information we need with `compiler_id` and `compiler_version`
+    // so that we can support compilers that insert pre-amble code even in `-E` mode
+    let test = b"#if defined(__NVCC__) && defined(__NVCOMPILER)
```

Comment:
Would it make sense to move the code into a new line so that all the relevant stuff is at the same indentation level?
```suggestion
    let test = b"
#if defined(__NVCC__) && defined(__NVCOMPILER)
```

Reply:
The current layout is what `cargo fmt` requires.

Reply:
Originaly misread the suggestion. Yes we can do this now that we aren't parsing only the first line.